amd_openvx_extensions/amd_rpp/source/tensor/PythonFunction.cpp

```
@@ -0,0 +1,315 @@
+
+// Map RpptDataType -> OpenVX type enum
+vx_enum getVxDataType(RpptDataType dataType) {
+    switch (dataType) {
+        case RpptDataType::F32:
+            return VX_TYPE_FLOAT32;
+        case RpptDataType::F16:
+            return VX_TYPE_FLOAT16;
+        case RpptDataType::U8:
+            return VX_TYPE_UINT8;
+        case RpptDataType::I8:
+            return VX_TYPE_INT8;
+        case RpptDataType::I16:
+            return VX_TYPE_INT16;
+        default:
+            throw std::runtime_error("Unsupported RpptDataType");
+    }
+}
+
+size_t getItemSize(RpptDataType dataType) {
+    switch (dataType) {
+        case RpptDataType::F32:
+            return sizeof(float);
+        case RpptDataType::F16:
+            return sizeof(vx_float16);
+        case RpptDataType::U8:
+            return sizeof(uint8_t);
+        case RpptDataType::I8:
+            return sizeof(int8_t);
+        case RpptDataType::I16:
+            return sizeof(int16_t);
+        default:
+            return 0;
+    }
+}
+
+struct PythonFunctionLocalData {
+    vx_uint32 deviceType;
+    RppPtr_t pSrc;
+    RppPtr_t pDst;
+    RpptGenericDescPtr pSrcGenericDesc;
+    RpptGenericDescPtr pDstGenericDesc;
+    vxTensorLayout inputLayout;
+    vxTensorLayout outputLayout;
+    size_t inputTensorDims[RPP_MAX_TENSOR_DIMS];
+    size_t outputTensorDims[RPP_MAX_TENSOR_DIMS];
+
+    // Bridge info
+    uint64_t function_id;
+    rocal_process_python_function_fn bridge_fn;
+};
+
+static vx_status VX_CALLBACK refreshPythonFunction(vx_node node, const vx_reference *parameters, vx_uint32 num, PythonFunctionLocalData *data) {
+    // CPU-only initial implementation: GPU not supported yet
+    if (data->deviceType == AGO_TARGET_AFFINITY_GPU) {
+        return VX_ERROR_NOT_IMPLEMENTED;
+    }
+
+    vx_status status = VX_SUCCESS;
+    STATUS_ERROR_CHECK(vxQueryTensor((vx_tensor)parameters[0], VX_TENSOR_BUFFER_HOST, &data->pSrc, sizeof(data->pSrc)));
```

In case the HIP is passed we need to raise an error code in this file, Please make that condition check somewhere

---

The HIP check is added in process function